distribution/pull_v2.go

```diff
@@ -32,7 +32,10 @@ import (
 	"golang.org/x/net/context"
 )
 
-var errRootFSMismatch = errors.New("layers from manifest don't match image configuration")
+var (
+	errRootFSMismatch  = errors.New("layers from manifest don't match image configuration")
+	errMediaTypePlugin = errors.New("plugins cannot be pulled in this manner. Use `docker plugin install`")
```

Comment:
What about

    target is a plugin manifest. Use `docker plugin install` to install plugins.